Move SymDispatchMode to its own file

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112038
* #112037
* #112036
* __->__ #112035

This is just code movement + a getter and a setter to break the
dependency of SymDispatchMode, and in turn, ProxySymDispatchMode on
sympy.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

lezcano committed:
Move SymDispatchMode to its own file

This is just code movement + a getter and a setter to break the
dependency of SymDispatchMode, and in turn, ProxySymDispatchMode on
sympy.

[ghstack-poisoned]

lezcano committed:
Update on "Remove dead code"


This dictionary is not used anywhere. The _make_dupe_guard function does
not exist anymore

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

lezcano committed:
Update on "Move SymDispatchMode to its own file"


This is just code movement + a getter and a setter to break the
dependency of SymDispatchMode, and in turn, ProxySymDispatchMode on
sympy.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

lezcano committed:
Update on "Move SymDispatchMode to its own file"


This is just code movement + a getter and a setter to break the
dependency of SymDispatchMode, and in turn, ProxySymDispatchMode on
sympy.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

lezcano committed:
Update on "Move SymDispatchMode to its own file"


This is just code movement + a getter and a setter to break the
dependency of SymDispatchMode, and in turn, ProxySymDispatchMode on
sympy.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

lezcano committed:
Update on "Move SymDispatchMode to its own file"


This is just code movement + a getter and a setter to break the
dependency of SymDispatchMode, and in turn, ProxySymDispatchMode on
sympy.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

lezcano committed:
Update on "Move SymDispatchMode to its own file"


This is just code movement + a getter and a setter to break the
dependency of SymDispatchMode, and in turn, ProxySymDispatchMode on
sympy.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

peterbell10 commented on torch/fx/experimental/symbolic_shapes.py:
```diff
@@ -144,16 +110,15 @@ def create_contiguous(shape):
     return list(reversed(strides))
 
 def _handle_sym_dispatch(func, args, kwargs):
```
Any reason not to put this in `_sym_dispatch_mode.py`?

lezcano replied:
I wanted to make changes that self-contained as possible. Of course this was before I had to move 1k lines into its own file. I'll move this one now.

lezcano committed:
Update on "Move SymDispatchMode to its own file"


This is just code movement + a getter and a setter to break the
dependency of SymDispatchMode, and in turn, ProxySymDispatchMode on
sympy.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

lezcano committed:
Update on "Move SymDispatchMode to its own file"


This is just code movement + a getter and a setter to break the
dependency of SymDispatchMode, and in turn, ProxySymDispatchMode on
sympy.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

lezcano committed:
Update on "Move SymDispatchMode to its own file"


This is just code movement + a getter and a setter to break the
dependency of SymDispatchMode, and in turn, ProxySymDispatchMode on
sympy.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

peterbell10 commented on torch/fx/experimental/_sym_dispatch_mode.py:
```diff
@@ -0,0 +1,58 @@
+    global SYM_FUNCTION_MODE
+    mode = sym_function_mode()
+    assert mode
+    SYM_FUNCION_MODE = mode.inner
```
Typo
```suggestion
    SYM_FUNCTION_MODE = mode.inner
```

peterbell10 commented on torch/fx/experimental/_sym_dispatch_mode.py:
```diff
@@ -0,0 +1,58 @@
+        types: List[Type] = []
+        return mode.__sym_dispatch__(func, types, args, kwargs)
+    finally:
+        SYM_FUNCION_MODE = mode
```
```suggestion
        SYM_FUNCTION_MODE = mode
```

lezcano committed:
Update on "Move SymDispatchMode to its own file"


This is just code movement + a getter and a setter to break the
dependency of SymDispatchMode, and in turn, ProxySymDispatchMode on
sympy.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

lezcano committed:
Update on "Move SymDispatchMode to its own file"


This is just code movement + a getter and a setter to break the
dependency of SymDispatchMode, and in turn, ProxySymDispatchMode on
sympy.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]
